amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/HierarchicalShardSyncerTest.java

```diff
@@ -1368,6 +1389,38 @@ public void testDetermineNewLeasesToCreateGraphBNoInitialLeasesTrim() {
                 .collect(Collectors.toList());
         final Set<ExtendedSequenceNumber> checkpoint = new HashSet<>(checkpoints);
 
+        final Set<String> expectedLeaseKeys = new HashSet<>(Arrays.asList("shardId-3", "shardId-4", "shardId-5"));
+        final Set<ExtendedSequenceNumber> expectedCheckpoint = new HashSet<>(
+                Collections.singletonList(ExtendedSequenceNumber.TRIM_HORIZON));
+
+        assertThat(newLeases.size(), equalTo(expectedLeaseKeys.size()));
+        assertThat(checkpoints.size(), equalTo(expectedLeaseKeys.size()));
+        assertThat(leaseKeys, equalTo(expectedLeaseKeys));
+        assertThat(checkpoint, equalTo(expectedCheckpoint));
+    }
+
+//    /**
+//     * Test CheckIfDescendantAndAddNewLeasesForAncestors (initial position TrimHorizon)
+//     * For shard graph B (see the construct method doc for structure).
+//     *
+//     * Current leases: empty set
+//     */
+    @Test
+    public void testDetermineNewLeasesToCreateGraphBNoInitialLeasesTrim() {
+        final List<Shard> shards = constructShardListForGraphB();
+        final List<Lease> currentLeases = new ArrayList<>();
+
+        final HierarchicalShardSyncer.LeaseSynchronizer emptyLeaseTableSynchronizer =
+                new HierarchicalShardSyncer.EmptyLeaseTableSynchronizer();
+
+        final List<Lease> newLeases = HierarchicalShardSyncer.determineNewLeasesToCreate(emptyLeaseTableSynchronizer,
```

Comment:
Basicalluy we can use the partialLTSynchronizer here as well, to see if this covers the empty lease table case as well. We know that empty lease table is simple case and can work.

Reply:
Won't the behavior be different depending on whether lease table is empty or non-empty?

For empty lease table, we still want to create leases for all shards right? So the expected leases would be shard-0, shard-1, ... shard-9, shard-10. For non-empty lease table, we use the single-level optimization so the expected leases would only be shard-0 and shard-1.

The single-level optimization is only implemented in the PR for the NonEmptyLeaseTableSynchronizer. Did we want this implemented for EmptyLeaseTableSynchronizer as well?